Cython/Compiler/Nodes.py

```diff
@@ -2965,10 +2966,12 @@ def analyse_declarations(self, env):
             # staticmethod() was overridden - not much we can do here ...
             self.is_staticmethod = False
 
-        if env.is_py_class_scope:
-            if self.name == '__new__':
+        if env.is_py_class_scope or env.is_c_class_scope:
+            if self.name == '__new__' and env.is_py_class_scope:
                 self.is_staticmethod = True
-            elif not self.is_classmethod and self.name in IMPLICIT_CLASSMETHODS:
+            elif self.name == INHERITANCE_HOOK and env.is_c_class_scope:
+                error(self.pos, "inheritance argument is not supported by extension class")
```

Comment:
I'd make this more explicit: `"__init_subclass__ is not supported by extensions class"`